contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -394,10 +399,14 @@ process_sign_request(struct sshbuf* request, struct sshbuf* response, struct age
 				RegQueryValueExW(sub, L"pin", 0, NULL, NULL, &epin_len) == 0) {
 				if (provider)
 					free(provider);
-				if (pin)
+				if (pin) {
+					SecureZeroMemory(pin, sizeof(pin));
```

Comment:
Should this be strlen() instead of sizeof()? Note that `pin`, `epin` might not always be NUL-terminated by the time we free the pointers.

Reply:
Good catch. And your observation regarding the possibility of `pin` and `epin` not being NULL-terminated by the time we free them is correct.
To address that I've switched to using `pin_len` and `epin_len`, which should always have the right values when we free the strings:
Within the loop, if acquiring the length failed in the previous iteration - execution flow will go to `done` (except for the first iteration, in which case there's nothing to free), hence we should always have the right values.
In the `done` section, those lengths will always have the right values unless we got there because the allocation for one of those strings failed - in which case the pointer will be NULL and we won't attempt to free it.
I've applied this to other instances where we use `SecureZeroMemory()` as well.